build/AnalyzeDependencies/AnalyzeDepsJS.ps1

```diff
@@ -0,0 +1,261 @@
+      version = $PkgInfo.Ver;
+      type    = "internal";
+      deps    = $InternalDeps
+    }
+  }
+
+  $PkgIds = $DumpData.Keys | ForEach-Object ToString
+  foreach ($PkgId in $PkgIds) {
+    foreach ($Dep in $DumpData[$PkgId].deps) {
+      $DepId = $Dep.name + ":" + $Dep.version
+      if (-Not $DumpData.ContainsKey($DepId)) {
+        $DumpData[$DepId] = @{
+          name    = $Dep.name;
+          version = $Dep.version;
+          type    = "internalbinary";
+          deps    = @()
+        }
+      }
+    }
+  }
+
+  return $DumpData
+}
+
+# Analyze package dependencies
+$Pkgs = @{ }
+$Deps = @{ }
+Resolve-Path $PackagesPath
+
+foreach ($PkgFile in Resolve-Path $PackagesPath) {
+  $PackageJson = Get-PackageJson $PkgFile
+  $LibraryName = $PackageJson.name
+  $LibraryVer = $PackageJson.version
+
+  Write-Host $LibraryName
+
+  $Pkgs[$LibraryName] = @{ Ver = $LibraryVer; Src = $PkgFile; Deps = New-Object System.Collections.ArrayList }
+  $PkgDeps = @{ }
+  
+  $Dependencies = (&{If($PackageJson.dependencies) {($PackageJson.dependencies | Get-Member -MemberType NoteProperty | Select Definition)} Else {$null}})
+  
+  foreach ($Dep in $Dependencies) {
+    $Depid = $Dep.Definition.Split(' =')[1];
+    $Depversion = $Dep.Definition.Split(' =')[2];
+    Save-PkgDep $PkgDeps "" $Depid $Depversion
+    if ($Deps.Count) {
+      $DepsKeys = ($Deps.Keys).Clone();
+      foreach ($TargetFramework in $DepsKeys) {
+        Save-Dep $Deps $TargetFramework $Depid $Depversion $LibraryName
+      }
+      Save-Dep $Deps "(others)" $Depid $Depversion $LibraryName
+    } else {
+      Save-Dep $Deps "(any)" $Depid $Depversion $LibraryName
+    }
+  }
+  
+  foreach ($DepName in $PkgDeps.Keys) {
+    $DepVersions = $PkgDeps[$DepName]
+    if ($DepVersions.Count -gt 1) {
+      foreach ($DepVersion in $DepVersions.Keys) {
+        $TargetFrameworks = $DepVersions[$DepVersion]
+        $Pkgs[$LibraryName]["Deps"].Add(@{name = $DepName; version = $DepVersion; label = ($TargetFrameworks -Join ", ") }) | Out-Null
+      }
+    } else {
+      $Pkgs[$LibraryName]["Deps"].Add(@{name = $DepName; version = ($DepVersions.Keys | Select-Object -first 1) }) | Out-Null
+    }
+  }
+}
+
+Write-Host "Analyzing $($Pkgs.Count) packages..."
+
+## Analyze lockfile
+$Locked = @{ }
+#if ($LockfilePath) {
+#  [xml]$PackageProps = Get-Content $LockfilePath
+#  foreach ($ItemGroup in $PackageProps.Project.ItemGroup) {
+#    if ($ItemGroup.Condition) {
+#      $Condition = $ItemGroup.Condition
+#    } else {
+#      $Condition = ""
+#    }
+#    foreach ($Entry in $ItemGroup.PackageReference) {
+#      if ($Entry.Update) {
+#        Save-Locked $Locked $Entry.Update $Entry.Version $Condition
+#      }
+#    }
+#  }
+#  Write-Host "Discovered $($Locked.Count) versions pinned in the lockfile."
+#} else {
+#  Write-Warning "No lockfile was provided, or the lockfile was empty. Declared dependency versions were not able to be validated against the lockfile."
+#}
```

Comment:
Do we need this?

Reply:
No. That remained from the DotNet version of AnalyzeDeps.ps1. It presented the possibility of a future enhancement. But it works without them, so I will delete them.